systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAuthorizationST.java

```diff
@@ -222,76 +231,70 @@ void testClusterVerification() {
-
-        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
-            kafka.getSpec().getKafka().getListeners().setExternal(
-                new KafkaListenerExternalNodePortBuilder()
-                    .withNewKafkaListenerAuthenticationOAuth()
-                        .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
-                        .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
-                        .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
-                        .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
-                        .withUserNameClaim(keycloakInstance.getUserNameClaim())
-                        .withTlsTrustedCertificates(
-                            new CertSecretSourceBuilder()
-                                .withSecretName(SECRET_OF_KEYCLOAK)
-                                .withCertificate(CERTIFICATE_OF_KEYCLOAK)
-                                .build())
-                        .withDisableTlsHostnameVerification(true)
-                    .endKafkaListenerAuthenticationOAuth()
-                    .build());
-
-            kafka.getSpec().getKafka().setAuthorization(
-                new KafkaAuthorizationKeycloakBuilder()
-                    .withClientId(KAFKA_CLIENT_ID)
-                    .withDisableTlsHostnameVerification(true)
-                    .withDelegateToKafkaAcls(false)
-                    // ca.crt a tls.crt
-                    .withTlsTrustedCertificates(
-                        new CertSecretSourceBuilder()
-                            .withSecretName(SECRET_OF_KEYCLOAK)
-                            .withCertificate(CERTIFICATE_OF_KEYCLOAK)
-                            .build()
-                    )
-                    .withTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
-                    .build());
-        });
-
-        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+        teamAOauthClientJob = new KafkaOauthClientsResource(
+                TEAM_A_PRODUCER_NAME,
+                TEAM_A_CONSUMER_NAME,
+                KafkaResources.tlsBootstrapAddress(CLUSTER_NAME),
+                TOPIC_A,
+                MESSAGE_COUNT,
+                "",
+                "a-consumer_group",
+                TEAM_A_CLIENT,
+                TEAM_A_CLIENT_SECRET,
+                keycloakInstance.getOauthTokenEndpointUri());
+
+        teamBOauthClientJob = new KafkaOauthClientsResource(
+                TEAM_B_PRODUCER_NAME,
+                TEAM_B_CONSUMER_NAME,
+                KafkaResources.tlsBootstrapAddress(CLUSTER_NAME),
+                TOPIC_A,
+                MESSAGE_COUNT,
+                "",
+                "x-" + ClientUtils.generateRandomConsumerGroup(),
+                TEAM_B_CLIENT,
+                TEAM_B_CLIENT_SECRET,
+                keycloakInstance.getOauthTokenEndpointUri());
```

Comment:
Another suggestion -> shouldn't be better to have builder for this? Just asking ... for the future PR?